bridge/bridge.go

```diff
@@ -22,7 +22,8 @@ type Config struct {
 
 	IRCServer        string
 	IRCServerPass    string
-	IRCListenerName  string // i.e, "DiscordBot", required to listen for messages in all cases
+	IRCListenerName  string            // i.e, "DiscordBot", required to listen for messages in all cases
+	IRCNickModes     map[string]string // Modes to set on each IRC connection @OnWelcome
```

Comment:
Have you tried an anonymous struct here, like this?

```
IRCNickModes struct {
    Add string
    Remove string
}
```

Reply:
I'll look into this soonish. I've got several other feature branches I need to clean up and submit pull requests for but after that's done I'll check if viper will do it and if so make use thereof.

Reply:
[Unmarshalling](https://github.com/spf13/viper#unmarshaling) is what you mean to look at? I would need to read more deeply and probably try some things but I get the feeling it would be applied to the entirety of the configuration... not just a single part?  If it has to be done for the entire config I believe that such an effort should be made into a separate pull request.

Reply:
Are we sure we want an anonymous struct, started work on this but you have to do the anonymous struct definition both in main.go and bridge.go files and seems like it's code duplication at that point. Can understand wanting unmarshaling and could still do that but not sure if an anonymous struct is the right way to go for it here.

Reply:
Seems an anonymous struct won't work:
```
./main.go:76:42: cannot use &ircNickModes (type *struct { Add string "mapstructure:\"add\""; Del string "mapstructure:\"del\"" }) as type viper.DecoderConfigOption in argument to viper.Unmarshal`
```
Going to use a regular struct.

Reply:
I was wrong, called the wrong function. it now uses an anonymous struct.